### PR TITLE
Adding two tests for 400ZR chromatic dispersion and eSNR

### PR DESCRIPTION
**Pull request by ahsaanyousaf**
Tests to ensure 400ZR chromatic dispersion and electrical signal to noise ratio is streamed correctly.

### CODE REVIEW

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+## Summary
+
+Validate 400ZR optics module reports Chromatic Dispersion.
+Chromatic Dispersion is frequency dependent change in signal phase velocity due
+to fiber measured in ps/nm 
+
+## Procedure
+
+*   Connect two ZR interfaces using a duplex LC fiber jumper such that TX
```
Comment (dplore) [inline feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:11]: In our lab, we usually have an optical switch in-between DUT ports and traffic generator ports.   Will this be ok for this test?  Any extra steps or changes in pass / fail criteria if we have this?

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD
Comment (dplore) [inline feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:1]: Is there any need to generate IP traffic for these tests?

File: feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD
Comment (dplore) [inline feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD:1]: Can we make this a subtest which is combined with the chromatic dispersion test?

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:1]: We don't need to generate IP traffic for these tests. There will be a separate test for IP traffic soak.

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD (hunk 1 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:11]: Do you know how much dB loss this switch path introduces in a setup like DUT-PortA = Switch = DUT-PortB
If the loss is 2 - 3dB we should be fine.

File: feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD:1]: Ack let me merge this with the other test

File: feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      * Both transceivers are tuned to a valid centre frequency
+        example 193.1 THz
+*   With the link ZR link established as explained above, verify that the
+    following ZR transceiver telemetry paths exist and are streamed for both
+    the ZR optics
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/instant
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/avg
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/min
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/max
+
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD:26]: Given what we just saw with one platform last week, for eSNR, CD, and other analog values we should also define what these paths should give when:
a) the device first boots up.
b) the transceiver is disabled.
c) the transceiver is enabled, but there is no signal (i.e. fiber break).

Curious on your thoughts, but for eSNR and CD, I think we could ask that (b) and (c) give a value of 0. The type here is Decimal64, so the most important part is that during these use cases, ONLY Decimal64 values are streamed. They should never stream a value such as string="nil" or string="inf". Especially, when the device is booting up, they should not stream until valid values can be streamed.

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD (hunk 1 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:11]: This b2b ZR link will be un-amplified so as long as we keep this below 3dB of insertion loss we should be fine

Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:11]: Updated

File: feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+**Note:** For min, max, and avg values, 10 second sampling is preferred. If 
+          10 seconds is not supported, the sampling interval used must be
+          communicated.
+
+
+*   Verify that the optics chromatic dispersion is updated after the interface flaps.
+
+    *   Enable a pair of ZR interfaces on the DUT as explained above.
+    *   Verify the ZR optics chromatic dispersion PM is in the normal range.
+    *   Disable or shut down the interface on the DUT.
```
[the remark this answers is not in the document]
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_chromatic_dispersion_test/README.MD:37]: Updated

File: feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD (hunk 2 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_electrical_signal_to_noise_ratio_test/README.MD:26]: Updated to capture these details

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+to fiber measured in ps/nm 
+
+## Procedure
+
+*   Connect two ZR interfaces using a duplex LC fiber jumper such that TX
+    output power of one is the RX input power of the other module. Connection
+    between the modules should pass through an optical switch that can be
+    controlled through automation to simulate a fiber cut.  
+*   To establish a point to point ZR link ensure the following:
+      * Both transceivers state is enabled
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:20]: grammar nit: "Both transceiver states are enabled."

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/instant
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/avg
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/min
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/max
+
+*   When the modules or the devices are still in a boot stage, they should not
+    stream any invalid string values like "nil" or "-inf" until valid values
+    are available for streaming.
+
+*   eSNR and CD streamed values should always be of type Decimal64.
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:41]: s/should/must

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    *   Verify the ZR optics eSNR and CD PM is updated to the value in the normal
+        range again.
+        * Typical expected value range for eSNR is 13.5 to
+          18 dB +/-0.1 dB.
+        * Typical CD expected value range is 0 to 2400 ps/nm.
+
+*   Verify that the optics eSNR and CD is updated after a fiber cut.
+
+    *   Enable a pair of ZR interfaces on the DUT as explained above.
+    *   Verify the ZR optics eSNR and CD PM is in the normal range.
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:65]: s/PM/telemetry values/

Please check throughout as "PM" feels a little like an outdated optical networking specific term that might not be widely known.

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+**Note:** For min, max, and avg values, 10 second sampling is preferred. If 
+          10 seconds is not supported, the sampling interval used must be
+          communicated.
+
+
+*   Verify that the optics eSNR and CD is updated after the interface flaps.
+
+    *   Enable a pair of ZR interfaces on the DUT as explained above.
+    *   Verify the ZR optics eSNR and CD PM is in the normal range.
+    *   Disable or shut down the interface on the DUT.
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:54]: Can you please call out to verify what values are expected during the fiber cut and that it should be a step to verify those?

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD (hunk 4 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:20]: Done

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD (hunk 6 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:65]: Done

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD (hunk 7 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:54]: Done

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/instant
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/avg
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/min
+    *   /terminal-device/logical-channels/channel/otn/state/esnr/max
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/instant
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/avg
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/min
+    *   /platform/components/component/optical-channel/state/chromatic-dispersion/max
+
+*   When the modules or the devices are still in a boot stage, they should not
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:37]: nit, s/should/must

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    stream any invalid string values like "nil" or "-inf" until valid values
+    are available for streaming.
+
+*   eSNR and CD streamed values must always be of type Decimal64.
+    When link interfaces are in down state 0 should be reported as a valid
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:42]: nit, s/should/must

**Review by ejbrever — APPROVED** [review]
LGTM, thanks.

File: feature/platform/transceiver/zr_esnr_and_cd_test/README.MD [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    *   Re-enable the optical switch connection to clear the fiber cut fault.
+    *   Verify the ZR optics eSNR and CD telemetry values are updated to the value in the normal
+        range again.
+        * Typical expected value range for eSNR is 13.5 to
+          18 dB +/-0.1 dB.
+        * Typical CD expected value range is 0 to 2400 ps/nm.
+
+## Config Parameter coverage
+
+*   /components/component/oc-transceiver:transceiver/oc-transceiver/config/enabled
```
Comment (dplore) [inline feature/platform/transceiver/zr_esnr_and_cd_test/README.MD:82]:
```suggestion
*   /components/component/transceiver/config/enabled
```

**Review by dplore — COMMENTED** [review]
Small correction to format of a path

**Review by dplore — APPROVED** [review]

**Pull request merged**